### ISSUE TITLE
AWS::StepFunctions::StateMachine - (JSONata) false-positive finding E3601 for Seconds in Wait state

### ISSUE THREAD

**Issue by ianmetcalf** [aws-cloudformation/cfn-lint#4342]
### CloudFormation Lint Version

1.43.3

### What operating system are you using?

Arch Linux

### Describe the bug

JSONata allows using a string for `Seconds` on a wait state as long as it evaluates to a number:

> A JSONata Wait State MAY specify a JSONata string for the "Seconds" field whose value, when evaluated, MUST be a non-negative integer. 

https://states-language.net/spec.html#jsonata-wait-state

Currently cfn-lint errors on this with `E3601 '...' is not of type 'number'`

### Expected behavior

Not to error

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09

Resources:
  StateMachineRole:
    Type: 'AWS::IAM::Role'
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - states.amazonaws.com
            Action:
              - 'sts:AssumeRole'

  StateMachine:
    Type: 'AWS::StepFunctions::StateMachine'
    Properties:
      RoleArn: !GetAtt StateMachineRole.Arn
      Definition:
        QueryLanguage: JSONata
        StartAt: Wait
        States:
          Wait:
            Type: Wait
            Seconds: '{% $number($states.input) %}'
            Next: Complete
          Complete:
            Type: Pass
            End: true
```
